query-node/mappings/src/content/app.ts

```diff
@@ -0,0 +1,123 @@
+    where: {
+      name: name,
+    },
+  })
+
+  if (isAppExists) {
+    logger.error('App already exists', { name })
+    return
+  }
+
+  const newApp = new App({
+    name,
+    id: appId,
+    createdById: memberId.toString(),
+    websiteUrl: appMetadata?.websiteUrl || undefined,
+    useUri: appMetadata?.useUri || undefined,
+    smallIcon: appMetadata?.smallIcon || undefined,
+    mediumIcon: appMetadata?.mediumIcon || undefined,
+    bigIcon: appMetadata?.bigIcon || undefined,
+    oneLiner: appMetadata?.oneLiner || undefined,
+    description: appMetadata?.description || undefined,
+    termsOfService: appMetadata?.termsOfService || undefined,
+    platforms: appMetadata?.platforms || undefined,
+    category: appMetadata?.category || undefined,
+    authKey: appMetadata?.authKey || undefined,
+  })
+  await store.save<CreateApp>(newApp)
+  logger.info('App has been created', { name })
+}
+
+export async function processUpdateAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IUpdateApp
+): Promise<void> {
+  const { appId, appMetadata } = message
+
+  const app = await getAppByIdAndMemberId(store, appId, memberId)
+
+  if (!app) {
+    logger.error("App doesn't exists or doesn't belong to the member", { appId, memberId: memberId.toString() })
+    return
+  }
+
+  app.websiteUrl = appMetadata?.websiteUrl || app.websiteUrl
+  app.useUri = appMetadata?.useUri || app.useUri
+  app.smallIcon = appMetadata?.smallIcon || app.smallIcon
+  app.mediumIcon = appMetadata?.mediumIcon || app.mediumIcon
+  app.bigIcon = appMetadata?.bigIcon || app.bigIcon
+  app.oneLiner = appMetadata?.oneLiner || app.oneLiner
+  app.description = appMetadata?.description || app.description
+  app.termsOfService = appMetadata?.termsOfService || app.termsOfService
+  app.platforms = appMetadata?.platforms || app.platforms
+  app.category = appMetadata?.category || app.category
+  app.authKey = appMetadata?.authKey || app.authKey
+
+  await store.save<App>(app)
+  logger.info('App has been updated', { appId })
+}
+
+export async function processDeleteAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IDeleteApp
+): Promise<void> {
+  const { appId } = message
+
+  const app = await getAppByIdAndMemberId(store, appId, memberId)
+
+  if (!app) {
+    logger.error("App doesn't exists or doesn't belong to the member", { appId, memberId: memberId.toString() })
+    return
+  }
+
+  await store.remove<App>(new App({ id: appId }))
+  logger.info('App has been removed', { appId })
+}
+
+async function getAppId(store: DatabaseManager, event: SubstrateEvent): Promise<string> {
+  let appId = `${event.blockNumber}-${event.indexInBlock}`
+  let tries = 0
+
+  // make sure app id is unique
+  while (await store.get<App>(App, { where: { id: appId } })) {
+    tries++
+    appId = `${event.blockNumber}-${event.indexInBlock}-${tries}`
+  }
+
+  return appId
+}
```

Comment:
This way of generating app IDs is only needed if entity ID input is required from user, e.g. in the creation of video category. Here, `let appId = ${event.blockNumber}-${event.indexInBlock}` would be sufficient to generate unique IDs every time since